src/time-range-picker/time-range-picker.tsx

```diff
@@ -0,0 +1,220 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import { TimeRangePickerHelper } from "./helper";
+import { TimeRangePickerDropdown } from "./timerangepicker-dropdown";
+import {
+    ArrowRangeIcon,
+    ArrowRight,
+    BottomHighlightEndTime,
+    BottomHighlightStartTime,
+    InputSelectorEndTimeElement,
+    InputSelectorStartTimeElement,
+    TimeContainer,
+    Wrapper,
+} from "./timerangepicker.styles";
+import { TimeRangePickerProps } from "./types";
+
+export const TimeRangePicker = ({
+    id,
+    disabled = false,
+    error,
+    value,
+    defaultValue,
+    placeholder,
+    format = "24hr",
+    readOnly,
+    onChange,
+    onBlur,
+    onSelectionCancel,
+    ...otherProps
+}: TimeRangePickerProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const [showStartTimeSelector, setShowStartTimeSelector] =
+        useState<boolean>(false);
+    const [showEndTimeSelector, setShowEndTimeSelector] =
+        useState<boolean>(false);
+    const [startTimeVal, setStartTimeVal] = useState("");
+    const [endTimeVal, setEndTimeVal] = useState("");
+
+    const nodeRef = useRef<HTMLDivElement>();
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+
+    useEffect(() => {
+        if (value) {
+            setStartTimeVal(value.startTime);
+            setEndTimeVal(value.endTime);
+        }
+    }, []);
+    useEffect(() => {
+        document.addEventListener("mousedown", handleMouseDownEvent);
+        document.addEventListener("keyup", handleKeyUpEvent);
+
+        return () => {
+            document.removeEventListener("mousedown", handleMouseDownEvent);
+            document.removeEventListener("keyup", handleKeyUpEvent);
+        };
+    }, [showEndTimeSelector]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleStartTimeFocus = useCallback(() => {
+        if (!disabled && !readOnly && !showStartTimeSelector) {
+            setShowEndTimeSelector(false);
+            setShowStartTimeSelector(true);
+        }
+    }, [showStartTimeSelector]);
+
+    const handleEndTimeFocus = useCallback(() => {
+        if (!disabled && !readOnly && !showEndTimeSelector) {
+            setShowStartTimeSelector(false);
+            setShowEndTimeSelector(true);
+        }
+    }, [showEndTimeSelector]);
```

Comment:
are the `useCallback`s needed? it seems like these event handlers need to access the updated state values anyway